src/topology/algebra/order/compact.lean

```diff
@@ -285,19 +313,38 @@ lemma continuous.bdd_above_range_of_has_compact_mul_support [has_one α]
   bdd_above (range f) :=
 @continuous.bdd_below_range_of_has_compact_mul_support (order_dual α) _ _ _ _ _ _ _ hf h
 
-/-- A continuous function is bounded below on a compact set. -/
-lemma is_compact.bdd_below_image {f : β → α} {K : set β}
-  (hK : is_compact K) (hf : continuous_on f K) : bdd_below (f '' K) :=
+lemma is_compact.continuous_Sup {f : γ → β → α}
+  {K : set β} (hK : is_compact K) (hf : continuous ↿f) :
+    continuous (λ x, Sup (f x '' K)) :=
 begin
-  rcases eq_empty_or_nonempty K with rfl|h, { rw [image_empty], exact bdd_below_empty },
-  obtain ⟨c, -, hc⟩ := hK.exists_forall_le h hf,
-  refine ⟨f c, _⟩, rintro _ ⟨x, hx, rfl⟩, exact hc x hx
+  rcases eq_empty_or_nonempty K with rfl|h0K,
+  { simp_rw [image_empty], exact continuous_const },
+  rw [continuous_iff_continuous_at],
+  intro x,
+  obtain ⟨y, hyK, h2y, hy⟩ :=
+    hK.exists_Sup_image_eq_and_ge h0K
+      (show continuous (λ y, f x y), from hf.comp $ continuous.prod.mk x).continuous_on,
+  rw [continuous_at, h2y, tendsto_order],
+  have := tendsto_order.mp ((show continuous (λ x, f x y), from
+    hf.comp $ continuous_id.prod_mk continuous_const).tendsto x),
+  refine ⟨λ z hz, _, λ z hz, _⟩,
+  { refine (this.1 z hz).mono (λ x' hx', hx'.trans_le $ le_cSup _ $ mem_image_of_mem (f x') hyK),
+    refine hK.bdd_above_image (hf.comp $ continuous.prod.mk x').continuous_on },
```

Comment:
Is there a reason why the second of these is a `refine` and not `exact`? Just seems odd.

Reply:
No, I usually start with refine, and then add more subterms, and I forget to turn it back into `exact`.